src/app_service/service/keycard/service.nim

```diff
@@ -174,10 +178,14 @@ QtObject:
   proc cancelCurrentFlow*(self: Service) =
     if self.busy:
       return
-    let response = keycard_go.keycardCancelFlow()
-    self.currentFlow = KCSFlowType.NoFlow
-    if self.doLogging:
-      debug "keycardCancelFlow", kcServiceCurrFlow=($self.currentFlow), response=response
+    # This code must not be executed on Darwin amd64
+    # since go 1.21 Intel macs are not able to handle nil cStrings from go
+    # resulting in bad flushGen 12 in prepareForSweep; sweepgen 0
+    if not (status_const.IS_MACOS and status_const.IS_INTEL):
+      let response = keycard_go.keycardCancelFlow()
+      self.currentFlow = KCSFlowType.NoFlow
+      if self.doLogging:
+        debug "keycardCancelFlow", kcServiceCurrFlow=($self.currentFlow), response=response
```

Comment:
```suggestion
    # This code must not be executed on Darwin amd64
    # since go 1.21 Intel macs are not able to handle nil cStrings from go
    # resulting in bad flushGen 12 in prepareForSweep; sweepgen 0
    if status_const.IS_MACOS and status_const.IS_INTEL:
      return
    let response = keycard_go.keycardCancelFlow()
    self.currentFlow = KCSFlowType.NoFlow
    if self.doLogging:
      debug "keycardCancelFlow", kcServiceCurrFlow=($self.currentFlow), response=response
```

Reply:
Also (if we keep this fix as is) would be nice to create a `status.const.SUPPORTS_KEYCARD` constant and keep your comment and link to this issue there. Will be simpler to track later.

Reply:
@igor-sirotin : I like the idea of early return and flipping the condition. Makes it more clear and explicit on why we return.